File: VB/MyDateEdit/MyVistaDateEditPainter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Controls
Imports System.Diagnostics

Namespace WindowsFormsApplication2
	Public Class MyVistaDateEditPainter
		Inherits VistaDateEditPainter
		Public Sub New(ByVal calendar As DateEditCalendarBase)
			MyBase.New(calendar)

		End Sub

		Public Overrides Sub DrawArrows(ByVal vdi As DevExpress.XtraEditors.ViewInfo.VistaDateEditInfoArgs)
			MyBase.DrawArrows(vdi)
			Dim args As MyVistaDateEditInfoArgs = TryCast(vdi, MyVistaDateEditInfoArgs)
			args.RightYearArrowInfo.Cache = args.Cache
			args.LeftYearArrowInfo.Cache = args.Cache
			ButtonPainter.DrawObject(args.RightYearArrowInfo)
			ButtonPainter.DrawObject(args.LeftYearArrowInfo)
		End Sub
	End Class
End Namespace

File: CS/MyDateEdit/MyVistaDateEditPainter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Controls;
using System.Diagnostics;

namespace WindowsFormsApplication2
{
    public class MyVistaDateEditPainter : VistaDateEditPainter
    {
        public MyVistaDateEditPainter(DateEditCalendarBase calendar)
            : base(calendar)
        {

        }

        public override void DrawArrows(DevExpress.XtraEditors.ViewInfo.VistaDateEditInfoArgs vdi)
        {
            base.DrawArrows(vdi);
            MyVistaDateEditInfoArgs args = vdi as MyVistaDateEditInfoArgs;
            args.RightYearArrowInfo.Cache = args.Cache;
            args.LeftYearArrowInfo.Cache = args.Cache;
            ButtonPainter.DrawObject(args.RightYearArrowInfo);
            ButtonPainter.DrawObject(args.LeftYearArrowInfo);
        }
    }
}
